File: app/src/main/java/com/example/edulearn/MainActivity.java
package com.example.edulearn;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.example.edulearn.api.ApiClient;
import com.example.edulearn.databinding.ActivityMainBinding;
import com.example.edulearn.utils.TokenManager;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private TokenManager tokenManager;
    private NavController navController;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        
        // Initialize TokenManager
        tokenManager = new TokenManager(this);
        
        // Initialize ApiClient with TokenManager
        ApiClient.initialize(tokenManager);
        
        // Setup Navigation
        navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        
        // Setup Bottom Navigation
        BottomNavigationView bottomNav = binding.bottomNavigation;
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.navigation_home, R.id.navigation_courses, 
                R.id.navigation_assignments, R.id.navigation_profile)
                .build();
        
        NavigationUI.setupWithNavController(bottomNav, navController);
        
        // Hide bottom navigation on auth screens
        navController.addOnDestinationChangedListener((controller, destination, arguments) -> {
            if (destination.getId() == R.id.loginFragment || 
                destination.getId() == R.id.registerFragment) {
                bottomNav.setVisibility(View.GONE);
            } else {
                bottomNav.setVisibility(View.VISIBLE);
            }
        });
        
        // Check if user is logged in and navigate accordingly
        if (!tokenManager.hasToken()) {
            navController.navigate(R.id.loginFragment);
        }
    }
    
    @Override
    public boolean onSupportNavigateUp() {
        return navController.navigateUp() || super.onSupportNavigateUp();
    }
}

File: app/src/main/java/com/example/edulearn/MainActivity.kt
package com.example.edulearn

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.example.edulearn.api.ApiClient
import com.example.edulearn.databinding.ActivityMainBinding
import com.example.edulearn.utils.TokenManager

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var tokenManager: TokenManager
    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Initialize TokenManager
        tokenManager = TokenManager(this)

        // Initialize ApiClient with TokenManager
        ApiClient.initialize(tokenManager)

        // Setup Navigation
        val navHostFragment = supportFragmentManager
            .findFragmentById(R.id.nav_host_fragment) as NavHostFragment
        navController = navHostFragment.navController

        // Setup Bottom Navigation
        binding.bottomNavigation.setupWithNavController(navController)

        // Hide bottom navigation on auth screens and show appropriate items based on user role
        navController.addOnDestinationChangedListener { _, destination, _ ->
            // Hide on auth screens
            if (destination.id == R.id.loginFragment ||
                destination.id == R.id.registerFragment) {
                binding.bottomNavigation.visibility = View.GONE
            } else {
                binding.bottomNavigation.visibility = View.VISIBLE

                // Update menu based on user role
                updateBottomNavigationMenu()
            }
        }

        // Check if user is logged in and navigate accordingly
        if (!tokenManager.getToken().isNullOrEmpty()) {
            navigateBasedOnRole()
        } else {
            navController.navigate(R.id.loginFragment)
        }
    }

    private fun updateBottomNavigationMenu() {
        val role = tokenManager.getUserRole()

        // Clear all menu items
        binding.bottomNavigation.menu.clear()

        // Add appropriate menu items based on role
        when (role) {
            "ADMIN" -> {
                binding.bottomNavigation.inflateMenu(R.menu.bottom_navigation_menu)
            }
            "TEACHER" -> {
                binding.bottomNavigation.inflateMenu(R.menu.bottom_navigation_menu)
            }
            else -> {
                binding.bottomNavigation.inflateMenu(R.menu.bottom_navigation_menu)
            }
        }
    }

    private fun navigateBasedOnRole() {
        // For now, navigate to home fragment
        // In a real implementation, you would navigate to different dashboards based on role
        navController.navigate(R.id.homeFragment)
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp() || super.onSupportNavigateUp()
    }
}
